### ISSUE TITLE
WAL panic if client has sent an empty measurement key

### ISSUE THREAD

**Issue by jipperinbham** [influxdata/influxdb#4206]
I realize this is more of a bad client/application but I would somewhat hope the database would reject any invalid data.

For reference: 3 node cluster (0.9.4.1) running on EC2 (r3.xlarge)

We were sending about 4-5k stats/second and things appeared to be running great but when the WAL would flush and every node would crash and go into a continuous crash loop. I've left out most of the gigantic stack trace as I think the most important bit was right at the top.

```
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 | [wal] 2015/09/22 23:56:35 Flush due to memory. Flushing 1 series with 927248 bytes from partition 3. Compacting 52277 series
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 | panic: error writing the wal to the index: write: key=, err=create series bucket: bucket name required
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 |
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 | goroutine 64 [running]:
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 | github.com/influxdb/influxdb/tsdb/engine/wal.(*Partition).flushAndCompact(0xc208234b60, 0x1, 0x0, 0x0)
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 |     /root/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:1060 +0x696
Sep 22 23:56:35 metrics-3 influxdb8: influxdb.1 | github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).triggerAutoFlush(0xc2080a81c0)
```

**Comment by otoolep** [comment]
Obviously we can't be sure without more examination, but we have pretty good data input validation, with extensive testing of the line protocol at a unit and integration level.

This may actually be a bug related to moving data from the WAL to BoltDB.

cc @pauldix

**Comment by jipperinbham** [comment]
prior to "fixing" the client application, I could run "SHOW MEASUREMENTS" and the first key return was an empty string FWIW

**Comment by jwilder** [comment]
This should be fixed in `0.10.0` and current nightlies.
